packages/flutter_test/test/finders_test.dart

```diff
@@ -390,6 +395,600 @@ void main() {
+          of: find.semantics.byLabel('label4'),
+          matching: find.semantics.byAction(SemanticsAction.copy),
+        );
+
+        try {
+          expect(finder, findsExactly(3));
+        } on TestFailure catch (e) {
+          failure = e;
+        }
+
+        expect(failure.message, contains('Actual: _AncestorSemanticsFinder:<Found 2 SemanticsNodes with action "SemanticsAction.copy" that are ancestors of leaf SemanticsNodes with label "label4"'));
+      });
+    });
+
+    group('descendant', () {
+      testWidgets('finds matching descendant nodes', (WidgetTester tester) async {
+        await tester.pumpWidget(semanticsTree);
+
+        final FinderBase<SemanticsNode> finder = find.semantics.descendant(
+          of: find.semantics.byLabel('label4'),
+          matching: find.semantics.byAction(SemanticsAction.copy),
+        );
+
+        expect(finder, findsOne);
+      });
+
+      testWidgets('fails with descriptive message', (WidgetTester tester) async {
+        late TestFailure failure;
+        await tester.pumpWidget(semanticsTree);
+
+        final FinderBase<SemanticsNode> finder = find.semantics.descendant(
+          of: find.semantics.byLabel('label4'),
+          matching: find.semantics.byAction(SemanticsAction.copy),
+        );
+
+        try {
+          expect(finder, findsNothing);
+        } on TestFailure catch (e) {
+          failure = e;
+        }
+
+        expect(failure.message, contains('Actual: _DescendantSemanticsFinder:<Found 1 SemanticsNode with action "SemanticsAction.copy" descending from root SemanticsNode with label "label4"'));
+      });
+    });
+
+    group('byPredicate', () {
+      testWidgets('finds nodes matching given predicate', (WidgetTester tester) async {
+        final RegExp replaceRegExp = RegExp(r'^[^\d]+');
+        await tester.pumpWidget(semanticsTree);
+
+        final SemanticsFinder finder = find.semantics.byPredicate(
+          (SemanticsNode node) {
+            final int labelNum = int.tryParse(node.label.replaceAll(replaceRegExp, '')) ?? -1;
+            return labelNum > 1;
+          },
+        );
+
+        expect(finder, findsExactly(4));
+      });
+
+      testWidgets('fails with default message', (WidgetTester tester) async {
+        late TestFailure failure;
+        final RegExp replaceRegExp = RegExp(r'^[^\d]+');
+        await tester.pumpWidget(semanticsTree);
+
+        final SemanticsFinder finder = find.semantics.byPredicate(
+          (SemanticsNode node) {
+            final int labelNum = int.tryParse(node.label.replaceAll(replaceRegExp, '')) ?? -1;
+            return labelNum > 1;
+          },
+        );
+        try {
+          expect(finder, findsExactly(5));
+        } on TestFailure catch (e) {
+          failure = e;
+        }
+
+        expect(failure.message, contains('Actual: _PredicateSemanticsFinder:<Found 4 matching semantics predicate (Closure: (SemanticsNode) => bool)'));
```

Comment:
Including the closure in the string seems kinda not very useful.... Do we do this elsewhere? Can we get a better description in here? Or if not, just omit it?

Reply:
This behavior is copied from the original `PredicateFinder` as a default if no description is provided. I agree that it's not _that_ useful, but I couldn't think of anything better. If you have any ideas, I'd be happy to update both, but since the predicate is passed to the Finder, it makes it pretty hard to have a useful description without the user also passing it in.

Reply:
Should we just omit this then?

Reply:
Removed in both predicate finders.